conans/model/editable_cpp_info.py

```diff
@@ -1,73 +1,73 @@
 # coding=utf-8
-import ntpath
 import os
-import posixpath
-import six
-from collections import defaultdict
+from collections import OrderedDict
+
 from six.moves import configparser
 
-from conans.client.tools.files import load
+from conans.errors import ConanException
+
+DEFAULT_LAYOUT_FILE = "default"
+LAYOUTS_FOLDER = 'layouts'
+
+
+def get_editable_abs_path(path, cwd, cache_folder):
+    # Check the layout file exists, is correct, and get its abs-path
+    if path:
+        layout_abs_path = path if os.path.isabs(path) else os.path.normpath(os.path.join(cwd, path))
+        if not os.path.isfile(layout_abs_path):
+            layout_abs_path = os.path.join(cache_folder, LAYOUTS_FOLDER, path)
+        if not os.path.isfile(layout_abs_path):
+            raise ConanException("Couldn't find layout file: %s" % path)
+        EditableCppInfo.load(layout_abs_path)  # Try if it loads ok
+        return layout_abs_path
+
+    # Default only in cache
+    layout_abs_path = os.path.join(cache_folder, LAYOUTS_FOLDER, DEFAULT_LAYOUT_FILE)
+    if os.path.isfile(layout_abs_path):
+        EditableCppInfo.load(layout_abs_path)
+        return layout_abs_path
 
 
 class EditableCppInfo(object):
-    WILDCARD = "*"
     cpp_info_dirs = ['includedirs', 'libdirs', 'resdirs', 'bindirs']
 
-    def __init__(self, data, uses_namespace):
+    def __init__(self, data):
         self._data = data
-        self._uses_namespace = uses_namespace
-
-    @classmethod
-    def load(cls, filepath, require_namespace):
-        """ Returns a dictionary containing information about paths for a CppInfo object: includes,
-        libraries, resources, binaries,... """
 
+    @staticmethod
+    def load(filepath):
         parser = configparser.ConfigParser(allow_no_value=True)
         parser.optionxform = str
-        parser.read(filepath)
+        try:
+            parser.read(filepath)
+        except configparser.Error:
+            raise ConanException("Error parsing layout file: %s" % filepath)
+        data = OrderedDict()
+        for section in parser.sections():
+            pkg, key = section.split(":", 1) if ':' in section else (None, section)
+            if key not in EditableCppInfo.cpp_info_dirs:
+                raise ConanException("Wrong cpp_info field '%s' in layout file: %s"
+                                     % (key, filepath))
+            data.setdefault(pkg, {})[key] = [k for k, _ in parser.items(section)]
```

Comment:
We are calling this function to test the file format, does it make sense to test the contents? We should check that every item is a valid folder after substituting the placeholders (maybe in a different function).